--- exercises/beginner/ex00.cairo	2022-05-10 13:54:32.000000000 +0200
+++ .solutions/ex00.cairo	2022-05-10 13:56:26.000000000 +0200
@@ -10,6 +10,9 @@
 # https://starknet.io/documentation/contracts/#contracts_storage
 # https://www.cairo-lang.org/docs/hello_starknet/intro.html
 # https://www.cairo-lang.org/docs/hello_starknet/user_auth.html#storage-maps
+@storage_var
+func dust(address : felt) -> (amount : felt):
+end
 
 # This code block define an `external` function
 # It can be called by other contracts (wallet or other)
