diff --git a/exercises/beginner/ex02.cairo b/exercises/beginner/ex02.cairo
index 77b34a0..116cf20 100644
--- a/exercises/beginner/ex02.cairo
+++ b/exercises/beginner/ex02.cairo
@@ -18,0 +19,4 @@ from starkware.cairo.common.math import assert_le
+struct Star:
+    member name : felt
+    member size : felt
+end
@@ -25,0 +30,3 @@ end
+@storage_var
+func star(address : felt, slot : felt) -> (size : Star):
+end
@@ -47,0 +55,15 @@ end
+@external
+func light_star{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(new_star : Star):
+    let (address) = get_caller_address()
+    let (current_amount) = dust.read(address)
+    assert_le(new_star.size, current_amount)
+    let (next_slot) = slot.read(address)
+
+    dust.write(address, current_amount - new_star.size)
+    star.write(address, next_slot, new_star)
+    slot.write(address, next_slot + 1)
+
+    a_star_is_born.emit(address, next_slot, new_star)
+
+    return ()
+end
@@ -67,0 +90,7 @@ end
+@view
+func view_star{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    address : felt, slot : felt
+) -> (star : Star):
+    let (res) = star.read(address, slot)
+    return (res)
+end
