--- exercises/beginner/ex03.cairo	2022-05-11 14:05:48.000000000 +0200
+++ .tmp_exercises/beginner/test_ex03.cairo	2022-05-11 14:04:26.000000000 +0200
@@ -58,8 +58,13 @@
 
     # TODO
     # Write a stop condition
+    if array_len == 0:
+        return ()
+    end
     # Insert the star at index 0 of the array
+    insert_star(address, array[0])
     # recursively call `batch_create_stars`
+    batch_create_stars(address, array_len - 1, array + Star.SIZE)
 
     return ()
 end
