--- exercises/strings/strings00.cairo	2022-05-20 18:37:19.510948914 +0200
+++ .solutions/strings00.cairo	2022-05-20 18:37:13.110928901 +0200
@@ -12,7 +12,9 @@
 # TODO: Fix the say_hello function by returning the appropriate short strings
 
 func say_hello() -> (hello_string : felt, hello_felt : felt, hello_hex : felt):
-    # FILL ME
+    let hello_string = 'Hello Starklings'
+    let hello_felt = '#L2-2022 #L3-2023'
+    let hello_hex = 'buidl buidl buidl'
     return (hello_string, hello_felt, hello_hex)
 end
 
