--- a/exercises/strings/strings01.cairo	2022-05-24 16:32:08.050760341 +0200
+++ b/exercises/strings/strings01.cairo	2022-05-24 15:41:30.545171320 +0200
@@ -11 +11,3 @@
-    let key = 0
+
+    # let key = 34642338947246174141841020580919298249066932857229157879292816894
+    let key = 'Twinkle Twinkle Little Star' - 'Another One Bites The Dust'
@@ -19 +21,2 @@
-    let ciphertext = 0
+    # let ciphertext = 3042032262124705672547337926850636172917305208722716617195516092604696795127
+    let ciphertext = ('Magic Starknet Money' - 0xc0de) / 1337
