--- exercises/registers/registers00.cairo	2022-05-16 10:23:49.000000000 -0400
+++ .tmp_solution/registers00.cairo	2022-05-18 11:38:32.000000000 -0400
@@ -10,15 +10,16 @@
 # Rewrite this function body in a high level syntax
 @external
 func ret_42() -> (r: felt):
-    [ap] = 42; ap++
-    ret
+    return (42)
 end
 
 # TODO
 # Rewrite this function body in a low level syntax, using registers
 @external
 func ret_0_and_1() -> (zero: felt, one: felt):
-    return (0, 1)
+    [ap] = 0; ap++
+    [ap] = 1; ap++
+    ret
 end
 
 #########
