--- exercises/registers/registers02.cairo	2022-06-11 22:20:43.000000000 -0400
+++ ./registers02.cairo	2022-06-11 22:18:21.000000000 -0400
@@ -17,3 +17,3 @@
-    # [ap] = 42; ap++
-    # [ap - 1] = 42
-    # [ap - 1] = 21
+    tempvar x = 42
+    assert x = 42
+    assert x = 21
@@ -27 +27 @@
-    # assert number = 42
+    [fp -3] = 42
@@ -37,0 +38 @@
+    assert [p_number] = 42
@@ -45,0 +47 @@
+    assert 42 = [p_number]
