--- exercises/registers/registers02.cairo	2022-05-18 11:40:28.000000000 -0400
+++ .tmp_solution/registers02.cairo	2022-05-18 11:40:22.000000000 -0400
@@ -8,29 +8,15 @@
 # Rewrite those functions with a high level syntax
 @external
 func sum_array(array_len: felt, array: felt*) -> (sum: felt):
-    [ap] = [fp - 4]; ap ++
-    [ap] = [fp - 3]; ap ++
-    [ap] = 0; ap ++
-    call rec_sum_array
-    ret
+    return rec_sum_array(array_len, array, 0)
 end
 
 func rec_sum_array(array_len: felt, array: felt*, sum: felt) -> (sum: felt):
-    jmp continue if [fp - 5] != 0
-
-    stop:
-    [ap] = [fp -3]; ap++
-    jmp done
-
-    continue:
-    [ap] = [[fp -4]]; ap++
-    [ap] = [fp - 5] - 1; ap++
-    [ap] = [fp - 4] + 1; ap++
-    [ap] = [ap - 3] + [fp - 3]; ap++
-    call rec_sum_array
-
-    done:
-    ret
+    if array_len == 0:
+        return (sum)
+    end
+    
+    return rec_sum_array(array_len - 1, array + 1, sum + array[0])
 end
 
 # TODO
@@ -38,12 +24,36 @@
 # It's possible to do it with only registers, labels and conditional jump. No reference or localvar
 @external
 func max{range_check_ptr}(a: felt, b: felt) -> (max: felt):
-    let (res) = is_le(a, b)
-    if res == 1:
-        return (b)
-    else:
-        return (a)
-    end
+    # Push arguments to the stack
+    [ap] = [fp - 5]; ap++ # range_check_ptr
+    [ap] = [fp - 4]; ap++ # a
+    [ap] = [fp - 3]; ap++ # b
+    
+    # This call will return two values
+    # 1) the updated range_check_ptr
+    # 2) 0 or 1 depending on which of a and b is greater
+    call is_le
+    
+    # Push return values to the stack
+    # There is two of them to push: range_check_ptr and max
+
+    # Push the first one, the updated range_check_ptr, onto the stack
+    [ap] = [ap - 2]; ap++
+
+    # Conditional jump
+    # The following blocks are an assembly level equivalent of the if/else pattern
+    jmp b_is_more if [ap - 2] !=0 # here [ap-2] is the second value returned by is_le, our boolean
+    
+    # Push either a or b to the stack 
+    a_is_more:
+    [ap] = [fp - 4]; ap++
+    jmp done
+    
+    b_is_more:
+    [ap] = [fp -3]; ap++
+    
+    done:
+    ret
 end
 
 #########
