--- ./exercises/operations/felt02.cairo	2022-08-07 14:50:40.505599191 +0200
+++ ./.solutions/felt02.cairo	2022-08-07 14:39:03.902994827 +0200
@@ -12,0 +13 @@
+    let res = 67 + x * (45 + x * (23 + x))
@@ -19,0 +21,6 @@
+    [ap] = x; ap++
+    [ap] = [ap - 1] + 23; ap++
+    [ap] = [ap - 1] * [ap - 2]; ap++
+    [ap] = [ap - 1] + 45; ap++
+    [ap] = [ap - 1] * [ap - 4]; ap++
+    [ap] = [ap - 1] + 67; ap++
